connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionCleanRoomE2ESuite.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import org.scalatest.concurrent.Eventually._
+
+import org.apache.spark.sql.connect.client.util.RemoteSparkSession
+import org.apache.spark.util.ThreadUtils
+
+/**
+ * Warning: SPARK-43648 moves two test cases related to `interrupt all` from `ClientE2ETestSuite`
+ * to the current test class to avoid the maven test issue of missing
+ * `org.apache.spark.sql.connect.client.SparkResult` during udf deserialization in server side. So
+ * please don't import classes that only exist in `spark-connect-client-jvm.jar` into the this
+ * class, as it will trigger similar maven test failures again.
+ */
+class SparkSessionCleanRoomE2ESuite extends RemoteSparkSession {
+
+  test("interrupt all - background queries, foreground interrupt") {
+    val session = spark
+    import session.implicits._
+    implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+    val q1 = Future {
+      spark.range(10).map(n => { Thread.sleep(30000); n }).collect()
```

Comment:
So just because ClientE2ETestSuite imported `org.apache.spark.sql.connect.client.SparkResult`, the serialization of this closure, which in no way uses SparkResult was referencing it, and causing an issue on the server during deserialization?
Wouldn't this pop up as a serious and hard to understand issue for users of Spark Connect - when they have their client code, they create a simple UDF closure in a class that imports/uses something totally unrelated to the UDF... and then UDF somehow pulls it in causing inexplicable deserialization problems on the server?

Also, this import of SparkResult was only added yesterday in https://github.com/apache/spark/commit/62338ed6cd9fba8bb92ec11cea643077e4b69db4#diff-7fa161b193c8792c8c0d8dd4bcae3e683ab8553edafa2ae5c13df42b26f612b0R41, while this test was already failing before, complaining about the lack of SparkResult class during deserialization? So it must have pulled it somehow even before it was imported in the test suite class?

Reply:
I am more inclined towards implicit imports caused by the use of `df.withResult`:

https://github.com/apache/spark/blob/64855fa55821332c4913ad06b2cf902d4c565a94/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala#L3181-L3186

I have tried deleting cases related to `withResult` in `ClientE2ETestSuite `, and these two use cases can also be passed

Reply:
> Wouldn't this pop up as a serious and hard to understand issue for users of Spark Connect - when they have their client code, they create a simple UDF closure in a class that imports/uses something totally unrelated to the UDF... and then UDF somehow pulls it in causing inexplicable deserialization problems on the server?

@vicennial do we have a way to clean up unused Imports during the udf serialization process? I haven't investigated this yet.

Reply:
I'm not an expert on this, but it looks to me that Spark has a `ClosureCleaner` that looks like something used for that purpose.
I think `ClosureCleaner` solves a similar problem between Spark driver and executors - makes irrelevant code not be tied up into closures that are send from the driver to executors for execution?
Maybe just doing `ClosureCleaner.clean` on `UdfPacket.function` will fix this?

Reply:
No, I've already tried it before this give this pr, at least the existing `ClosureCleaner ` cannot achieve the goal

Reply:
:-(

Reply:
@rednaxelafx could I ask for your help?
You're the last person who wrote a lot of code to ClosureCleaner (in 2020).
Do you understand better what might be going on, and why the serialization of this closure in `UdfPacket.function` would pull in unrelated stuff that `ClosureCleaner` wouldn't clean?

Reply:
Crash or threw NoSuchFieldException or throw other exceptions?  In my impression, `ClosureCleaner` focuses on the repl, and there are some reflection calls base on convention rules in the code